### PR TITLE
Package canary.0.0.1

### PR DESCRIPTION
**Pull request by chrismamo1**
### `canary.0.0.1`
Capture unhandled exceptions and automatically report them through various channels
Developed by the University of Zurich, Canary is a tool for capturing unhandled exceptions and automatically reporting them to your development team.

---
* Homepage: https://github.com/chrismamo1/canary
* Source repo: git+https://github.com/chrismamo1/canary.git
* Bug tracker: https://github.com/chrismamo1/canary/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

File: packages/canary/canary.0.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.08.0"}
+  "lwt"
+  "ptime"
+  "cohttp"
+  "cohttp-lwt"
+  "cohttp-lwt-unix"
+  "yojson"
+  "lwt_ppx"
```
Comment (mseri) [inline packages/canary/canary.0.0.1/opam:21]:
```suggestion
  "lwt_ppx" {>= "1.2.2"}
```
Pre `dune` ppxs are often not working nicely with dune. This is the version in which `lwt_ppx` was ported to done

File: packages/canary/canary.0.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "ppx_deriving_yojson"
```
Comment (mseri) [inline packages/canary/canary.0.0.1/opam:22]:
```suggestion
  "uri" {>= "4.0.0"}
  "ppx_deriving_yojson"
```
` ``Generic` was introduced in version 4:
```
# File "lib/canary.ml", line 64, characters 50-58:
# 64 |               let enc = Uri.pct_encode ~component:`Generic in
#                                                        ^^^^^^^^
# Error: This expression has type [> `Generic ]
#        but an expression was expected of type Uri.component
#        The second variant type does not allow tag(s) `Generic
```

**Comment by mseri** [comment]
I don't know how to fix the remaining lower bound. @kit-ty-kate any idea?

**Comment by kit-ty-kate** [comment]
Thanks! Sorry for the wait. Could you return the fixes to the opam file upstream so they can be taken into account for next time?

**Pull request merged**